File: workout-zio/src/main/scala/workout/domain/AcceptedResponse.scala
package workout.domain

import zio.json._
import java.util.UUID

final case class AcceptedResponse(
    success: Boolean,
    correlationId: UUID,
    message: String
)

object AcceptedResponse {
  implicit val encoder: JsonEncoder[AcceptedResponse] = DeriveJsonEncoder.gen[AcceptedResponse]
  implicit val decoder: JsonDecoder[AcceptedResponse] = DeriveJsonDecoder.gen[AcceptedResponse]

  def apply(correlationId: UUID): AcceptedResponse =
    AcceptedResponse(
      success = true,
      correlationId = correlationId,
      message = "Request accepted for processing"
    )
}

File: workout-spring-webflux/src/main/java/com/workout/app/api/dto/AcceptedResponse.java
package com.workout.app.api.dto;

import java.util.UUID;

public record AcceptedResponse(boolean success, UUID correlationId, String message) {
    public static AcceptedResponse ok(UUID correlationId) {
        return new AcceptedResponse(true, correlationId, "Request accepted for processing");
    }
}

